### PR TITLE
[GPII-3965] Delete old json files and metrics

### PR DESCRIPTION
**Pull request by amatas**
This PR is needed as a first step of the deployment of GPII-3965, it will delete all the alerts and lbm in order to let the next PR to create them using Terraform code.

### CODE REVIEW

**Comment by natarajaya** [comment]
@amatas How do you plan to deploy this?

I understand this is needed to avoid conflicts with new resources created in #496, but pushing this to prod alone would destroy all Stackdriver primitives and leave prod infra without any automated monitoring / alerting.

**Comment by amatas** [comment]
> @amatas How do you plan to deploy this?
> 
> I understand this is needed to avoid conflicts with new resources created in #496, but pushing this to prod alone would destroy all Stackdriver primitives and leave prod infra without any automated monitoring / alerting.

the #513 and #496 go all together, one after the other. If any of them fail the two would be reverted.

File: common/modules/gcp-stackdriver-monitoring/main.tf [hunk 1]
```diff
@@ -81,28 +74,16 @@ resource "null_resource" "apply_stackdriver_monitoring" {
...
       RETRIES=5
       RETRY_COUNT=1
+      unset STACKDRIVER_DEADLINE_EXCEEDED
       while [ "$STACKDRIVER_DEADLINE_EXCEEDED" != "false" ]; do
         STACKDRIVER_DEADLINE_EXCEEDED="false"
         echo "[Try $RETRY_COUNT of $RETRIES] Destroying Stackdriver resources..."
         bundle exec ruby -e '
           require "/rakefiles/stackdriver.rb"
-          destroy_resources({"alert_policies"=>[],"notification_channels"=>[]})
+          destroy_resources({"log_based_metrics"=>[]})
```
[the remark this answers is not in the document]
Comment (amatas) [reply common/modules/gcp-stackdriver-monitoring/main.tf]: This is going to be deleted in the next PR, I didn't pay too much attention to the refactoring of the code. Just only wanted to be sure that it is able to remove the old alerts.

**Comment by amatas** [comment]
> What is the upgrade path for individual `dev` clusters operated by our developers?

`rake destroy_module["k8s/stackdriver/monitoring"]`
`rake destroy_module["k8s/stackdriver/lbm"]`
(or remove the alerts and the lbm manually)
merge the latest master and run `rake`

**Comment by stepanstipl** [comment]
> rake destroy_module

Unfortunately, this won't work once you're on a version past this PR (and we always recommend devs to be on `master`). And even running `rake destroy` won't fix this.

I'm afraid that the upgrade story is quite hairy and will require a lot of support from the team to migrate each individual dev cluster. Have we considered how would alternatives, that might possibly not require any manual intervention, look like?

**Comment by natarajaya** [comment]
> Unfortunately, this won't work once you're on a version past this PR

I think this may be an issue.

@amatas What do you think about keeping Stackdriver ruby client and adding a script to destroy all resources properly, so developers have a way to migrate their envs?

**Comment by amatas** [comment]
> > Unfortunately, this won't work once you're on a version past this PR
> 
> I think this may be an issue.
> 
> @amatas What do you think about keeping Stackdriver ruby client and adding a script to destroy all resources properly, so developers have a way to migrate their envs?

This is done in https://github.com/gpii-ops/gpii-infra/pull/522/commits/c3e29dff1c39c22289678ab4b5e1d5f127499137 and https://github.com/gpii-ops/gpii-infra/pull/522/commits/f8d6c6a4c48b366ce319f84c4768071c00b9d0e8

File: common/modules/gcp-stackdriver-monitoring/main.tf [hunk 2]
```diff
@@ -43,28 +43,21 @@ resource "template_dir" "resources" {
...
       while [ "$STACKDRIVER_DEADLINE_EXCEEDED" != "false" ]; do
         STACKDRIVER_DEADLINE_EXCEEDED="false"
-        echo "[Try $RETRY_COUNT of $RETRIES] Applying Stackdriver resources..."
+        echo "[Try $RETRY_COUNT of $RETRIES] Destroying Stackdriver resources..."
         bundle exec ruby -e '
           require "/rakefiles/stackdriver.rb"
-          resources = read_resources("${path.module}/resources_rendered")
-          apply_resources(resources)
+          destroy_resources({"alert_policies"=>[],"notification_channels"=>[]})
+          destroy_uptime_checks(["${join("\",\"", google_monitoring_uptime_check_config.this.*.name)}"])
```
Comment (stepanstipl) [inline common/modules/gcp-stackdriver-monitoring/main.tf]: I think code might fail, as there's no `google_monitoring_uptime_check_config.this` resource in this module.

**Comment by stepanstipl** [comment]
@amatas can we close this one, since it's not required anymore, right? (part of #522)

**Comment by amatas** [comment]
Not needed at it is performed in #522

**Pull request closed without merging**